src/credentials/CHIPCert.h

```diff
@@ -653,6 +653,24 @@ CHIP_ERROR ConvertX509CertToChipCert(const uint8_t * x509Cert, uint32_t x509Cert
 CHIP_ERROR ConvertX509CertsToChipCertArray(const ByteSpan & x509NOC, const ByteSpan & x509ICAC, uint8_t * chipCertArrayBuf,
                                            uint32_t chipCertArrayBufSize, uint32_t & chipCertBufLen);
 
+/**
+ * @brief Extract NOC and ICA Certificates from a CHIP certificate array.
+ *        This function takes a CHIP certificate array and splits it into single
+ *        NOC and ICA CHIP certificates.
+ *
+ *        NOC certificate must be provided.
+ *        ICA certificate is optional. It will be omitted (nullptr, 0) if the CHIP certificate array contains only one entry.
+ *
+ *        The API enforces that the NOC is issued by ICA (if ICA is provided).
```

Comment:
How is that enforced? I don't see code to check chaining. That chaining should be validated at time of use, perhaps, and this comment line removed if the check is not added. I think that the check is probably a good check, but not easy to do right now.

Reply:
We do check if the TLV reached the end after parsing the NOC - Check lines 806-812 from CHIPCertFromX509.cpp. If it reaches the end, we set the ICAC as (nullptr, 0)

Reply:
Verifying presence/absence of NOC is not the same of `enforcing that the NOC is issued by ICA`. I see no parsing to validate that the AKID of the NOC matches the SKID of the ICA if present, nor issuer subject check, nor any signature verification of NOC from ICA. I strongly recommand removing this one comment about "enforces that the NOC is issued by ICA", since that check should be done after extracting both certs, during AddNOC/UpdateNOC processing, and not as part of a basic data access/getter operation.

Reply:
I removed the comment as per your suggestion.